ONSdigital/rm-action-service
# Add code formatter to maven build and relax checkstyle

# Motivation and Context
As a pull request reviewer
I want the code format to be consistent
So I can focus on reviewing code changes instead of formatting inconsistencies

# What has changed

Adding maven plugin to format the java code automatically on builds. The
style of the formatter is opinionated favouring googles formatting. The
benefit of this is it allows us to easily plugin in googles check style
file. The checkstyle.xml has been made more lenient to ignore things
like missing javadoc and abbreviations.

# How to test?
1. Checkout branch
1. Mess up format
1. Run `mvn install`
1. Check format is fixed

# Links
Formatter plugin: https://github.com/coveo/fmt-maven-plugin
Checkstyle plugin: https://github.com/checkstyle/checkstyle
Intellij formatter plugin: https://github.com/google/google-java-format#intellij
Eclipse formatter plugin: https://github.com/google/google-java-format#eclipse
https://trello.com/c/yXqb0VAK/116-reformat-java-services
